plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerConnectorTest.java

```diff
@@ -213,47 +216,48 @@ public void testColumnComment()
+                .isFullyPushedDown();
+
+        // varchar NOT IN with small compaction threshold
+        assertThat(query(
+                Session.builder(getSession())
+                        .setCatalogSessionProperty("sqlserver", "domain_compaction_threshold", "1")
+                        .build(),
+                "SELECT regionkey, nationkey, name FROM nation WHERE name NOT IN ('POLAND', 'ROMANIA', 'VIETNAM')"))
+                // no pushdown because it was converted to range predicate
                 .isNotFullyPushedDown(
                         node(
                                 FilterNode.class,
-                                // verify that pushed down constraint is applied by the connector
+                                // verify that no constraint is applied by the connector
                                 tableScan(
-                                        tableHandle -> {
-                                            TupleDomain<ColumnHandle> constraint = ((JdbcTableHandle) tableHandle).getConstraint();
-                                            ColumnHandle nameColumn = constraint.getDomains().orElseThrow()
-                                                    .keySet().stream()
-                                                    .map(JdbcColumnHandle.class::cast)
-                                                    .filter(column -> column.getColumnName().equals("name"))
-                                                    .collect(onlyElement());
-                                            return constraint.getDomains().get().get(nameColumn)
-                                                    .equals(Domain.multipleValues(
-                                                            createVarcharType(25),
-                                                            ImmutableList.of(
-                                                                    utf8Slice("POLAND"),
-                                                                    utf8Slice("ROMANIA"),
-                                                                    utf8Slice("VIETNAM"))));
-                                        },
+                                        tableHandle -> ((JdbcTableHandle) tableHandle).getConstraint().isAll(),
                                         TupleDomain.all(),
                                         ImmutableMap.of())));
 
+        // varchar IN without domain compaction
+        assertThat(query("SELECT regionkey, nationkey, name FROM nation WHERE name IN ('POLAND', 'ROMANIA', 'VIETNAM')"))
+                .matches("VALUES " +
+                        "(BIGINT '3', BIGINT '19', CAST('ROMANIA' AS varchar(25))), " +
+                        "(BIGINT '2', BIGINT '21', CAST('VIETNAM' AS varchar(25)))")
+                .isFullyPushedDown();
+
         // varchar IN with small compaction threshold
```

Comment:
And an additional test to ensure that `NOT IN` with small compaction threshold ? Sorry missed it last time.

Reply:
I think we have it : 
```
        // varchar NOT IN with small compaction threshold
        assertThat(query(
                Session.builder(getSession())
                        .setCatalogSessionProperty("sqlserver", "domain_compaction_threshold", "1")
                        .build(),
                "SELECT regionkey, nationkey, name FROM nation WHERE name NOT IN ('POLAND', 'ROMANIA', 'VIETNAM')"))
                // no pushdown because it was converted to range predicate
                .isNotFullyPushedDown(
                        node(
                                FilterNode.class,
                                // verify that no constraint is applied by the connector
                                tableScan(
                                        tableHandle -> ((JdbcTableHandle) tableHandle).getConstraint().isAll(),
                                        TupleDomain.all(),
                                        ImmutableMap.of())));
```

Reply:
Thank you